### PR TITLE
all: use quicktest for tests

### PR DESCRIPTION
**Pull request by mhilton**

### CODE REVIEW

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore-client/226/

File: cmd/charm/charmcmd/cmd_test.go
Comment (rogpeppe) [inline cmd/charm/charmcmd/cmd_test.go]: For readability purposes, I think it might be nice to add the cleanup after the thing it's cleaning up so it reads like a defer statement (and elsewhere).

File: cmd/charm/charmcmd/cmd_test.go
Comment (rogpeppe) [inline cmd/charm/charmcmd/cmd_test.go]: Let's please not call this "homeSuite" as that sounds like it's got a suite of tests, which it doesn't (it was always an aberration that the juju helper "suites" were named that way). Maybe:

	// setFakeHome sets up environment variables
	// to point to a new temporary home directory,
	// and installs cleanup functions to restore the
	// changed global state.
	func setFakeHome(c *qt.C) {
		oldHome := utils.Home()
		c.AddCleanup(func() {
			utils.SetHome(oldHome)
		})
		utils.SetHome(c.Mkdir())
		c.Setenv("HOME", utils.Home())
		err := os.MkdirAll(jujutesting.JujuXDGDataHomePath(), 0755)
		c.Assert(err, qt.Equals, nil)
		c.AddCleanup(func() {
			osenv.SetJujuXDGDataHome("")
		})
		osenv.SetJujuXDGDataHome(jujutesting.JujuXDGDataHomePath())
	}

?

File: cmd/charm/charmcmd/cmd_test.go
Comment (rogpeppe) [inline cmd/charm/charmcmd/cmd_test.go]: Similarly to homeSuite, it would be nice if this could be renamed to something not including the word "suite".
Maybe a similar pattern to what I suggested above might work here too? Maybe consider splitting it into domain-specific helpers too, if that makes sense.

I'm not sure whether embedding the common functionality makes as much sense now too - maybe put it in a named field. (that might be too disruptive, I guess, for this PR)

File: cmd/charm/charmcmd/suite_test.go
Comment (rogpeppe) [inline cmd/charm/charmcmd/suite_test.go]: I changed my mind about using regular interface conversion here. If someone defines an Init method with the wrong signature, I think we want it to panic rather than just ignoring it. YMMV.

File: dependencies.tsv
Comment (rogpeppe) [inline dependencies.tsv:4]: Looks like we're still dependent on juju/testing. Why?

**Review by rogpeppe — CHANGES_REQUESTED** [review]
This is turning out pretty well, I think. I've a bunch of comments, but the general idea is good. How about adding a `.travis.yml` file too, so we can check that travis tests are passing before we land it?

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore-client/227/

File: cmd/charm/charmcmd/cmd_test.go
Comment (rogpeppe) [inline cmd/charm/charmcmd/cmd_test.go]: I think that `setFakeHome` (`initFakeHome`?) might be a better name for this, as it indicates the nature of its side effect. *shrug*

File: cmd/charm/charmcmd/suite_test.go
Comment (rogpeppe) [inline cmd/charm/charmcmd/suite_test.go]: Maybe put this test outside the loop?

File: cmd/charm/charmcmd/suite_test.go
Comment (rogpeppe) [inline cmd/charm/charmcmd/suite_test.go]: I wonder if this should fail if there are no test methods defined in the suite?

File: cmd/charm/charmcmd/suite_test.go
Comment (rogpeppe) [inline cmd/charm/charmcmd/suite_test.go]: This doesn't look right to me. Have you tested this code?

File: cmd/charm/charmcmd/suite_test.go
Comment (rogpeppe) [inline cmd/charm/charmcmd/suite_test.go]: Or maybe just use st and don't bother with typeName?

**Review by rogpeppe — APPROVED** [review]
LGTM modulo a few comments and suggestions, thanks.

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore-client/228/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore-client/229/

**Comment by jujugui** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://ci-gce.jujugui.org:8080//job/charmstore-client/230/

**Pull request merged**
